### PR TITLE
[UT] [Portsyncd] Added Unit Tests for portsyncd

### PR DESCRIPTION
**Pull request by vivekrnv**
**What I did**

**Added Unit tests for portsyncd module:**
-    Create a new make target under mock_tests/Makefile.am to include a target for linksync.cpp
-    Mocked a rtnl_link object for testing onMsg method of Linksync.cpp
-    Used/updated the existing mock infra to allow for testing.

**Why I did it**

To increase code/feature coverage

**How I verified it**

```
vkarri@e4bc0a3e5630:/sonic$ make -f slave.mk target/debs/bullseye/swss_1.0.0_amd64.deb
..........
make[3]: Entering directory '/sonic/src/sonic-swss/tests'
Making check in mock_tests
make[4]: Entering directory '/sonic/src/sonic-swss/tests/mock_tests'
make  check-TESTS
make[5]: Entering directory '/sonic/src/sonic-swss/tests/mock_tests'
make[6]: Entering directory '/sonic/src/sonic-swss/tests/mock_tests'
PASS: tests_portsyncd
PASS: tests_intfmgrd
PASS: tests
============================================================================
Testsuite summary for sonic-swss 1.0
============================================================================
# TOTAL: 3
# PASS:  3
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================

```

```
vkarri@4897dcba242c:/sonic/src/sonic-swss/tests/mock_tests$ make tests_portsyncd 
.............
..............
..............
vkarri@5e69834fbb01:/sonic/src/sonic-swss/tests/mock_tests$ ./tests_portsyncd 
Running main() from /build/googletest-YnT0O3/googletest-1.10.0.20201025/googletest/src/gtest_main.cc
[==========] Running 6 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 6 tests from PortSyncdTest
[ RUN      ] PortSyncdTest.test_linkSyncInit
[       OK ] PortSyncdTest.test_linkSyncInit (0 ms)
[ RUN      ] PortSyncdTest.test_cacheOldIfaces
[       OK ] PortSyncdTest.test_cacheOldIfaces (0 ms)
[ RUN      ] PortSyncdTest.test_onMsgNewLink
[       OK ] PortSyncdTest.test_onMsgNewLink (0 ms)
[ RUN      ] PortSyncdTest.test_onMsgDelLink
[       OK ] PortSyncdTest.test_onMsgDelLink (0 ms)
[ RUN      ] PortSyncdTest.test_onMsgMgmtIface
[       OK ] PortSyncdTest.test_onMsgMgmtIface (0 ms)
[ RUN      ] PortSyncdTest.test_onMsgIgnoreOldNetDev
[       OK ] PortSyncdTest.test_onMsgIgnoreOldNetDev (0 ms)
[----------] 6 tests from PortSyncdTest (1 ms total)

[----------] Global test environment tear-down
[==========] 6 tests from 1 test suite ran. (1 ms total)
[  PASSED  ] 6 tests.

```

**Details if related**

### CODE REVIEW

**Review by keboliu — DISMISSED** [review]

**Comment by liat-grozovik** [comment]
@prsunny could you please help to review and signoff?

**Comment by prsunny** [comment]
This PR has conflict with https://github.com/Azure/sonic-swss/pull/2305. I suggest lets revisit this PR after 2305 is merged. The newly introduced helper file in this PR shall have to be revisited and I don't think it would be required.

**Comment by vivekreddynv** [comment]
> This PR has conflict with #2305. I suggest lets revisit this PR after 2305 is merged. The newly introduced helper file in this PR shall have to be revisited and I don't think it would be required.

The file will still be needed as the structures g_portSet, g_init and the method handlePortConfigFromConfigDB are still required, but yeah this can wait until #2305 is merged.

**Comment by prsunny** [comment]
> > This PR has conflict with #2305. I suggest lets revisit this PR after 2305 is merged. The newly introduced helper file in this PR shall have to be revisited and I don't think it would be required.
> 
> The file will still be needed as the structures g_portSet, g_init and the method handlePortConfigFromConfigDB are still required, but yeah this can wait until #2305 is merged.
For just one function, I wouldn't recommend creating new file. IMO, it is unnecessary and future cherry-picks have issue if there is a bug fix.

**Comment by liat-grozovik** [comment]
as we still didnt close the HLD of the other PR, i suggest we use this one first and once the other PR can be approved and get in it will have to be aligned with this one. Can we then merge this one and update the other.
@Junchao-Mellanox  fyi

**Comment by vivekreddynv** [comment]
> > > This PR has conflict with #2305. I suggest lets revisit this PR after 2305 is merged. The newly introduced helper file in this PR shall have to be revisited and I don't think it would be required.
> > 
> > 
> > The file will still be needed as the structures g_portSet, g_init and the method handlePortConfigFromConfigDB are still required, but yeah this can wait until #2305 is merged.
> > For just one function, I wouldn't recommend creating new file. IMO, it is unnecessary and future cherry-picks have issue if there is a bug fix.

TBH, future cherry-picks will be an issue nevertheless because i can't keep these three methods "g_portSet, g_init and the method handlePortConfigFromConfigDB" in portsyncd.cpp. If not for the helper file, i've to move them to Linksync.cpp, so any change to those can't be cherry-picked directly to older branches.

**Comment by liat-grozovik** [comment]
@vivekrnv could you please handle conflicts and followup on checkers?

**Comment by prsunny** [comment]
i dont think we should combine refactoring and portsyncd changes together. Can you please have this PR only for the portsyncd?

File: portsyncd/portsyncd_helper.cpp
Comment (prsunny) [inline portsyncd/portsyncd_helper.cpp:1]: I think we discussed it last time, please move this to portsyncd itself. I don't see a reason why it has to be part of helper file and cause merge conflicts for future bug fixes.

**Review by prsunny — APPROVED** [review]

**Pull request merged**
